### PR TITLE
String utils repl all

### PR DESCRIPTION
**Pull request by alexjhawk**

### CODE REVIEW

**Comment by alexjhawk** [comment]
Fixing code style. Dont submit review

File: src/com/hms_networks/americas/sc/string/StringUtils.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+   * Method to replace all instances of <code>before</code> with <code>after</code> in <code>string
+   * </code> and return the result.
+   *
+   * @param string string to modify
+   * @param before to replace
+   * @param after replacement
+   * @return string with replacement completed
+   * @since 1.0
+   */
+  public static String replaceAll(String string, String before, String after) {
```
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/string/StringUtils.java:21]:
```suggestion
  public static String replace(String string, String target, String replacement) {
```

Changes to match String's replace method from 1.5
https://docs.oracle.com/javase/7/docs/api/java/lang/String.html#replace(java.lang.CharSequence,%20java.lang.CharSequence)

Comment (TomKimsey) [reply src/com/hms_networks/americas/sc/string/StringUtils.java:21]: This method should probably also throw the null pointer exception that the 1.5 version does if any of the strings are null.

Comment (TomKimsey) [reply src/com/hms_networks/americas/sc/string/StringUtils.java:21]: This should not be named replaceAll as that would imply it takes a regex, replace is more appropriate to match the forbidden 1.5 code.

Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/string/StringUtils.java]: updated x2

Comment (TomKimsey) [reply src/com/hms_networks/americas/sc/string/StringUtils.java]: I think you missed the rename of before and after

Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/string/StringUtils.java]: yes i did

Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/string/StringUtils.java]: fixed

File: src/com/hms_networks/americas/sc/string/StringUtils.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    // Throw NullPointerException if any supplied strings are null
+    if (string == null || target == null || replacement == null) throw new NullPointerException();
+
+    // Create String buffer for building the modified string
+    StringBuffer stringBuffer = new StringBuffer();
+
+    // Loop through each character in -string-
+    for (int x = 0; x < string.length(); ) {
+      // Search for an instance of -before- in -string-
+      int indexOfBefore = string.indexOf(target, x);
```
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/string/StringUtils.java:32]:
```suggestion
      int indexOfTarget = string.indexOf(target, x);
```

File: src/com/hms_networks/americas/sc/string/StringUtils.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  public static String replace(String string, String target, String replacement) {
+    // Throw NullPointerException if any supplied strings are null
+    if (string == null || target == null || replacement == null) throw new NullPointerException();
+
+    // Create String buffer for building the modified string
+    StringBuffer stringBuffer = new StringBuffer();
+
+    // Loop through each character in -string-
+    for (int x = 0; x < string.length(); ) {
+      // Search for an instance of -before- in -string-
```
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/string/StringUtils.java:31]:
```suggestion
      // Search for an instance of -target- in -string-
```

File: src/com/hms_networks/americas/sc/string/StringUtils.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+      int indexOfBefore = string.indexOf(target, x);
+
+      // Perform replacement if at start index
+      if (x == indexOfBefore) {
```
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/string/StringUtils.java:35]:
```suggestion
      if (x == indexOfTarget) {
```

File: src/com/hms_networks/americas/sc/string/StringUtils.java [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+/**
+ * Class providing utilities for working with and manipulating {@link String} objects.
+ *
+ * @author HMS Networks, MU Americas Solution Center
+ * @since 1.1
+ */
+public class StringUtils {
+
+  /**
+   * Method to replace all instances of <code>before</code> with <code>after</code> in <code>string
```
Comment (TomKimsey) [inline src/com/hms_networks/americas/sc/string/StringUtils.java:12]:
```suggestion
   * Method to replace all instances of <code>target</code> with <code>replacement</code> in <code>string
```

Comment (alexjhawk) [reply src/com/hms_networks/americas/sc/string/StringUtils.java]: all fixed. Sorry for missing those.

**Pull request merged**